packages/eslint-plugin/src/rules/no-unnecessary-generics.ts

```diff
@@ -54,7 +54,7 @@ const rule = createRule({
               context.report({
                 data: { name },
                 messageId: "sole",
-                node: parserServices.tsNodeToESTreeNodeMap.get(res.soleUse),
```

Comment:
I think you should be able to drop `soleUse` or simplify otherwise, actually; I think it's only returned for this particular case.

Reply:
Lemme have a looksee - just grabbing some dinner

Reply:
simplified - think this is what you were suggesting?

Reply:
Yes, even further than I was suggesting, even! (Which is not bad.)

Reply:
Let's go even further 🔪 no need for an interface with just the one property. `type Result = "ok" | ...`!

Reply:
...or even `MessageId | undefined` 🔪 🔪

Reply:
gosh you guys are brutal.  Let me sharpen the knife

Reply:
how's this?

Reply:
Looks great!

> `feat: 🔪`

nice touch :)